### DISCUSSION TITLE
Re-Starting TWW-Wizard

### DISCUSSION THREAD

**Discussion by urskaufmann** [teksi/wastewater#184] [Ideas]
When digitizing reaches or manholes, I do often change to the identifier-tool or the vertex-tool to add attributes etc. The TWW Data Entry stays "open", but I can not change back to the digitizing mode with one click (or I don't know how). I have to change from wastewater Structure to Reach and back again (waiting for the saving) or I have to Stop Data Entry and Start Data Entry (waiting for the saving...).
I want a solution with one click on the wizard again to continue with digitizing

**Comment by ponceta** [comment]
I've a feeling that 
**waiting for the saving...**
Is the main issue here.
Is your saving taking time?

Otherwise it could be a feature request but I would be in favor of using standard QGIS tools instead.

There's a deprecation warning on this particular tool :
```
2024-03-12T09:01:47     WARNING    warning:C:\Users/pulpoar/AppData/Roaming/QGIS/QGIS3\profiles\pulpoar/python/plugins\teksi_wastewater\tools\twwmaptooladdfeature.py:232: DeprecationWarning: IndividualLayerSettings constructor is deprecated
              settings = QgsSnappingConfig.IndividualLayerSettings(
             
             traceback: File "C:\Users/pulpoar/AppData/Roaming/QGIS/QGIS3\profiles\pulpoar/python/plugins\teksi_wastewater\teksi_wastewater_plugin.py", line 383, in wizard
              self.wizarddock = TwwWizard(self.iface.mainWindow(), self.iface)
              File "C:\Users/pulpoar/AppData/Roaming/QGIS/QGIS3\profiles\pulpoar/python/plugins\teksi_wastewater\gui\twwwizard.py", line 54, in __init__
              self.mapToolAddReach = TwwMapToolAddReach(
              File "C:\Users/pulpoar/AppData/Roaming/QGIS/QGIS3\profiles\pulpoar/python/plugins\teksi_wastewater\tools\twwmaptooladdfeature.py", line 232, in __init__
              settings = QgsSnappingConfig.IndividualLayerSettings(
```

**Reply by urskaufmann** [reply]
Yes, waiting is the main problem. I decide, when I want to save, not the app. And if you look from user side: I want to continue digitizing a reach and I have first to click on stop data entry or an wastewater structure  - makes not sense

**Reply by ponceta** [reply]
Waiting is an issue, I want to raise any performance issue and solve these.

I open another issue for the performance aspects and keep this for the re-start digitizing functionnality.
